common/persistence/dataInterfaces.go

```diff
@@ -873,6 +873,11 @@ type (
 		PageSize int
 		// Token to continue reading next page of history append transactions.  Pass in empty slice for first page
 		NextPageToken []byte
+
+		// LastFirstTransactionID specified in mutable state. Only used for reading in reverse order.
+		LastFirstTransactionID int64
+		// Whether we want to return events in reverse order.
+		ReverseOrder bool
```

Comment:
why not making a dedicated API just for reverse the history?

when reading from start to end, the LastFirstTransactionID is not used at all, plus the processing logic is different

Reply:
+1 for adding a new persistence api for reverse read. imo it should make the code easier to read and follow.